app/controllers/sensors_controller.rb

```diff
@@ -59,6 +59,8 @@ def set_sensor
   def set_readings
     @final_readings = @sensor.sensor_readings.final.order(:created_at).last(50).reverse
     @debug_readings = @sensor.sensor_readings.debug.order(:created_at).last(50).reverse
+    @final_readings = @final_readings.group_by { |r| r.created_at.to_date }
```

Comment:
Is `created_at` the actual date? I always thought this was automatically set by `ActiveRecord` on creation of a new item. If that’s the case, that date might be different from the measurement date…